### PR TITLE
Apr 1293 2 1 people in control

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/PeopleInControl/PeopleInControlTests.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                    .Verifiable("view model not returned");
+
+
+            _controller = new RoatpGatewayPeopleInControlController(_contextAccessor.Object, _applyApiClient.Object, _logger.Object, _gatewayValidator.Object,  _orchestrator.Object);
+            }
+
+            [Test]
+            public void check_people_in_control_request_is_sent()
+            {
+                var pageId = "PeopleInControl";
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/PeopleInControl/PeopleInControlTests.cs:74]: Can this come from the GatewayPageIds constants

File: src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/PeopleInControl/PeopleInControlTests.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                var _result = _controller.GetGatewayPeopleInControlPage(applicationId, pageId).Result;
+                _orchestrator.Verify(x => x.GetPeopleInControlViewModel(It.IsAny<GetPeopleInControlRequest>()), Times.Once());
+            }
+
+        [Test]
+        public void post_people_in_control_happy_path()
+        {
+            var applicationId = Guid.NewGuid();
+            var pageId = "PeopleInControl";
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/PeopleInControl/PeopleInControlTests.cs:83]: Can this come from the GatewayPageIds constants

File: src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/PeopleInControl/PeopleInControlTests.cs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+            _applyApiClient.Verify(x => x.SubmitGatewayPageAnswer(It.IsAny<Guid>(), It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>()), Times.Once);
+            _orchestrator.Verify(x => x.GetPeopleInControlViewModel(It.IsAny<GetPeopleInControlRequest>()), Times.Once());
+        }
+
+        [Test]
+        public void post_people_in_control_path_with_errors()
+        {
+            var applicationId = Guid.NewGuid();
+            var pageId = "PeopleInControl2";
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/PeopleInControl/PeopleInControlTests.cs:107]: Can this come from the GatewayPageIds constants

File: src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/PeopleInControl/Orchestrator/PeopleInControlTests.cs [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+            };
+            _organisationSummaryApiClient.Setup(x => x.GetWhosInControlFromSubmitted(applicationId)).ReturnsAsync(whosInControlFromSubmitted);
+
+
+            var request = new GetPeopleInControlRequest(applicationId, UserName);
+            var response = _orchestrator.GetPeopleInControlViewModel(request);
+
+            var viewModel = response.Result;
+
+            Assert.AreEqual(GatewayPageIds.PeopleInControl, viewModel.PageId);
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/PeopleInControl/Orchestrator/PeopleInControlTests.cs:134]: There's a huge amount of asserts here, is there a better way? Like could you write an equality function or something?

File: src/SFA.DAS.AdminService.Web/Services/Gateway/PeopleInControlOrchestrator.cs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                RoatpGatewayConstants.Captions.PeopleInControlChecks,
+                RoatpGatewayConstants.Headings.PeopleInControl,
+                NoSelectionErrorMessages.PeopleInControlCheck);
+
+            _logger.LogInformation($"Retrieving people in control - type of organisation for application {request.ApplicationId}");
+            model.TypeOfOrganisation = await _organisationSummaryApiClient.GetTypeOfOrganisation(request.ApplicationId);
+
+
+            _logger.LogInformation($"Retrieving people in control - company directors for application {request.ApplicationId}");
+            model.CompanyDirectorsData = new PeopleInControlData
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Services/Gateway/PeopleInControlOrchestrator.cs:39]: Could the Captions and External Source Headings be put into a constant string or lookup somewhere?

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/PeopleInControl.cshtml [hunk 6]
```diff
@@ -0,0 +1,89 @@
+﻿@using SFA.DAS.AssessorService.Domain.Extensions
+@model SFA.DAS.AdminService.Web.ViewModels.Roatp.Gateway.PeopleInControlPageViewModel
+
+@{
+        ViewBag.Title = $"Gateway - {Model.Heading}";
+        Layout = "_Layout";
+        var peopleInControlPartialViewName = "~/Views/Roatp/Apply/Gateway/Pages/_PeopleInControlTable.cshtml";
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/PeopleInControl.cshtml:7]: This looks a bit odd. Everything else in inline 🤷‍♂

File: src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/PeopleInControl/Orchestrator/PeopleInControlTests.cs (hunk 4 above)
Comment (MarkFCain) [reply src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/PeopleInControl/Orchestrator/PeopleInControlTests.cs:134]: Maybe. I might just break each area into it's own test, and head the whole setup to do the run once, if that makes sense.  I'll change it whatever tho

**Pull request merged**